inc/updater/namespace.php

```diff
@@ -25,9 +27,47 @@ function bootstrap() {
  * @return void
  */
 function get_fair_document_data( $obj ) : void {
-	global $metadata, $release;
-	$metadata = $obj->metadata ?? $obj->package;
-	$release = $obj->release;
+	global  $release;
+
+	$packages = [];
+	// phpcs:disable HM.Security.NonceVerification.Recommended
+	if ( $obj instanceof Upgrader ) {
+		if ( isset( $_REQUEST['action'] ) && 'fair-install-plugin' === $_REQUEST['action'] ) {
+			$did = isset( $_REQUEST['id'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['id'] ) ) : '';
+			if ( $did === $obj->package->id ) {
+				$release[ $did ] = $obj->release;
+			}
+		}
+	}
```

Comment:
Optional: Could trim this down if we merge some conditions and save on entering the initial `if` at all.

```suggestion
	if (
		$obj instanceof Upgrader
		&& isset( $_REQUEST['action'], $_REQUEST['id'] )
		&& 'fair-install-plugin' === $_REQUEST['action']
	) {
		$did = sanitize_text_field( wp_unslash( $_REQUEST['id'] ) );
		if ( $did === $obj->package->id ) {
			$release[ $did ] = $obj->release;
		}
	}
```